### ISSUE TITLE
Calling `method()` on a test stub created using `createStubForIntersectionOfInterfaces()` throws an unexpected exception

### ISSUE THREAD

**Issue by hugok96** [sebastianbergmann/phpunit#5818]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 10.5.18
| PHP version         | 8.1.27
| Installation Method | Composer

#### Summary

Calling the ``method()`` method on a ``Stub`` created through the ``TestCase::createStubForIntersectionOfInterfaces()`` method throws an unexpected exception.

#### Current behavior

When calling the ``m̀ethod()`` method on a ``Stub`` created through the ``TestCase::createStubForIntersectionOfInterfaces()`` method, an exception is thrown: ``Error: Call to undefined method TestStub_Intersection_AInterface_f00d4e13_af77f05e::expects()``.

#### How to reproduce

```php
public function testExample(): void
{
    $stub = self::createStubForIntersectionOfInterfaces([FirstInterface::class, SecondaryInterface::class]);
    $stub->method('method');
}
```

#### Expected behavior

Calling ``method()`` on a ``Stub`` should not throw an exception.

**Comment by sebastianbergmann** [comment]
Thank you for patience and effort to open this new issue. It is now much clearer to me.

**Comment by sebastianbergmann** [comment]
According to git-bisect, this is broken since 073671d57c98a8d592d19a4a251997c28686f9a4 which was first released in PHPUnit 10.4.0.
